sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/throughputControl/controller/container/ThroughputContainerController.java

```diff
@@ -109,11 +109,11 @@ private ThroughputResolveLevel getThroughputResolveLevel(Set<ThroughputControlGr
     @SuppressWarnings("unchecked")
     public <T> Mono<T> init() {
         return this.resolveContainerMaxThroughput()
-            .flatMap(controller -> this.createAndInitializeGroupControllers())
-            .doOnSuccess(controller -> {
-                Schedulers.parallel().schedule(() -> this.refreshContainerMaxThroughputTask(this.cancellationTokenSource.getToken()).subscribe());
-            })
-            .thenReturn((T) this);
+                   .then(this.createAndInitializeGroupControllers())
```

Comment:
but for then, "this.createAndInitializeGroupControllers" will still be executed right? The intention here is to only execute when the previous step succeeded?